apisix/plugins/ai-proxy/drivers/openai.lua

```diff
@@ -0,0 +1,83 @@
+--
+-- Licensed to the Apache Software Foundation (ASF) under one or more
+-- contributor license agreements.  See the NOTICE file distributed with
+-- this work for additional information regarding copyright ownership.
+-- The ASF licenses this file to You under the Apache License, Version 2.0
+-- (the "License"); you may not use this file except in compliance with
+-- the License.  You may obtain a copy of the License at
+--
+--     http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+--
+local _M = {}
+
+local core = require("apisix.core")
+local http = require("resty.http")
+
+local pairs = pairs
+
+-- globals
+local DEFAULT_HOST = "api.openai.com"
+local DEFAULT_PORT = 443
+
+
+function _M.request(conf, request_table, ctx)
+    local httpc, err = http.new()
+    if not httpc then
+        return nil, "failed to create http client to send request to LLM server: " .. err
+    end
+    httpc:set_timeout(conf.timeout)
+
+    local custom_host = core.table.try_read_attr(conf, "override", "host")
+    local custom_port = core.table.try_read_attr(conf, "override", "port")
+    local custom_path = core.table.try_read_attr(conf, "override", "path")
+    local custom_scheme = core.table.try_read_attr(conf, "override", "scheme")
+
+    local ok, err = httpc:connect({
+        scheme = custom_scheme or "https",
+        host = custom_host or DEFAULT_HOST,
+        port = custom_port or DEFAULT_PORT,
+        ssl_verify = conf.ssl_verify,
+        ssl_server_name = custom_host or DEFAULT_HOST,
```

Comment:
Is it required, and should lua-resty-http automatically infer it?

Reply:
we should provide is compulsorily.

Reply:
I'm not aware of any other APISIX calls passing that parameter, I mean `ssl_server_name`, so I suspect it should be automatically inferred from the host, but no matter, it doesn't mess up the functionality and provides greater certainty, I'll take it.